docs/data/data-grid/rows/rows.md

```diff
@@ -17,18 +17,27 @@ Otherwise, the grid will re-apply heavy work like sorting and filtering.
 
 {{"demo": "RowsGrid.js", "bg": "inline"}}
 
-:::warning
-Each row object should have a field that uniquely identifies the row.
-By default, the grid will use the `id` property of the row. Note that [column definition](/x/react-data-grid/column-definition/) for `id` field is not required.
+## Row identifier
+
+Each row must have a unique id.
 
-When using dataset without a unique `id` property, you can use the `getRowId` prop to specify a custom id for each row.
+This id is used internally to identify the row in the various models (for instance the row selection model) and to track the row across updates.
+By default, the grid will use the `id` property of the row.
+
+When using a dataset without a unique `id` property, you can use the `getRowId` prop to specify the id of each row.
+This can be useful if your id is stored in another property like shown below:
 
 ```tsx
 <DataGrid getRowId={(row) => row.internalId} />
 ```
 
 {{"demo": "RowsGridWithGetRowId.js", "bg": "inline", "defaultCodeOpen": false}}
 
```

Comment:
```suggestion

If no such unique identifier exists in the data set, then you must create it by some other means, but this scenario should be avoided because it leads to issues with other features of the grid.

Note that it is not necessary to create a column to display the unique identifier data. 
The data grid pulls this information directly from the data set itself, not from anything that is displayed on the screen.

```